spirit/comment/forms.py

```diff
@@ -138,3 +138,34 @@ def save(self):
         name = default_storage.save(name, file)
         file.url = default_storage.url(name)
         return file
+
+
+class CommentFileForm(forms.Form):
+
+    file = forms.FileField()
+
+    def __init__(self, user=None, *args, **kwargs):
+        super(CommentFileForm, self).__init__(*args, **kwargs)
+        self.user = user
+
+    def clean_file(self):
+        file = self.cleaned_data['file']
+
+        # Probably not reliable, but better than no validations
+        # https://stackoverflow.com/questions/6460848/how-to-limit-file-types-on-file-uploads-for-modelforms-with-filefields
+        if file.content_type not in settings.ST_ALLOWED_UPLOAD_FILE_FORMAT:
```

Comment:
Just FYI, this is not better than no validation. The content type is supplied by the client. It's possible to upload a EXE file and tell the server the media type is a PDF for example. I think for Windows checking the extension is best. For linux and mac I dunno, prolly checking the file header type with something like [magic] . I'm not opposed to including it as a dependency.

Then we can have a setting and check both thinks. Like this:

```python
ST_FILE_ALLOWED_FILE_FORMAT = {
    'pdf': 'application/pdf',
}
```

[magic]: https://github.com/ahupp/python-magic

Reply:
You're right about the vulnerability, my mistake!

I'll be looking into python-magic. :+1: